packages/dx-grid-core/src/plugins/integrated-summary/computeds.ts

```diff
@@ -97,6 +97,17 @@ export const groupSummaryValues: GroupSummaryValuesFn = (
         row,
         rows: [],
       });
+      levelIndex = getLevelIndex(levelKey);
+    }
+    const isCollapsedNestedGroupRow = collapsedRows && levelIndex > 0;
+    if (!levelKey || isCollapsedNestedGroupRow) {
+      levels.forEach((level) => {
+        if (!levelKey) {
```

Comment:
The condition is checked on each iteration. Code can be replaced with something like this:
```js
const rowsToAppend = !levelKey ? [row] : collapsedRows;
levels.forEach((level) => {
  level.rows.push(...rowsToAppend);
});
```